src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -1,10 +1,16 @@
 from __future__ import annotations
 
+import sys
 import json
 import typing as t
 import logging
 import dataclasses
 
+if sys.version_info >= (3, 9, 2):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
```

Comment:
Since 3.7 will reach EOL in 2 months, it is probably best not to do this and just use `t.TypedDict` instead when BentoML drops 3.7 support

Reply:
Any reason this is checking for 3.9.2?

Reply:
Sorry, read the above comment. Looks like this will have to stay in even if we drop 3.7 support? Definitely not a reason to delay this PR, I think.

Reply:
Probably also need to maintain `typing_extensions` in pyproject.toml